src/ObjectUtils/PdbFileLlvm.cpp

```diff
@@ -214,6 +255,118 @@ PdbFileLlvm::PdbFileLlvm(std::filesystem::path file_path, const ObjectFileInfo&
+  llvm::pdb::PDBFile& pdb_file = native_session->getPDBFile();
+
+  if (!pdb_file.hasPDBDbiStream()) {
+    return ErrorMessage("PDB file does not have a DBI stream.");
+  }
+  llvm::Expected<llvm::pdb::DbiStream&> debug_info_stream = pdb_file.getPDBDbiStream();
+  ORBIT_CHECK(debug_info_stream);
+
+  if (!pdb_file.hasPDBTpiStream()) {
+    return ErrorMessage("PDB file does not have a TPI stream.");
+  }
+  llvm::Expected<llvm::pdb::TpiStream&> type_info_stream = pdb_file.getPDBTpiStream();
+  ORBIT_CHECK(type_info_stream);
+
+  llvm::FixedStreamArray<llvm::object::coff_section> section_headers =
+      debug_info_stream->getSectionHeaders();
+
+  std::vector<SymbolInfo> symbol_infos;
+  absl::flat_hash_set<uint64_t> addresses_from_module_debug_stream;
+  OUTCOME_TRY(LoadDebugSymbolsFromModuleStreams(
+      pdb_file, debug_info_stream.get(), type_info_stream.get(), section_headers, object_file_info_,
+      &symbol_infos, &addresses_from_module_debug_stream));
+
+  if (!pdb_file.hasPDBPublicsStream()) {
+    return ErrorMessage("PDB file does not have a public symbol stream.");
+  }
+  llvm::Expected<llvm::pdb::PublicsStream&> public_symbol_stream = pdb_file.getPDBPublicsStream();
+  ORBIT_CHECK(public_symbol_stream);
+
+  if (!pdb_file.hasPDBSymbolStream()) {
+    return ErrorMessage("PDB file does not have a symbol stream.");
+  }
+  llvm::Expected<llvm::pdb::SymbolStream&> symbol_stream = pdb_file.getPDBSymbolStream();
+  ORBIT_CHECK(symbol_stream);
+
+  LoadDebugSymbolsFromPublicSymbolStream(public_symbol_stream.get(), symbol_stream.get(),
+                                         section_headers, object_file_info_,
+                                         addresses_from_module_debug_stream, &symbol_infos);
+
+  // We try to find the missing size information from public symbols from the section contributions.
+  // Note, that we sometimes have multiple names for the same address, so we use a vector here as
+  // the map's value type.
+  absl::flat_hash_map<uint64_t, std::vector<SymbolInfo*>> address_to_symbols_with_missing_size{};
+  for (SymbolInfo& symbol_info : symbol_infos) {
+    if (symbol_info.size() != ObjectFile::kUnknownSymbolSize) {
+      continue;
+    }
+    address_to_symbols_with_missing_size[symbol_info.address()].push_back(&symbol_info);
+  }
+  SectionContributionsVisitor visitor{object_file_info_, &section_headers,
+                                      &address_to_symbols_with_missing_size};
+  debug_info_stream->visitSectionContributions(visitor);
+
+  // It does not seem to be guaranteed that we have section contribution information for all
+  // symbols, so let's try to deduce the size of the missing symbols based on the distance from the
+  // next symbol
+  DeduceDebugSymbolMissingSizesAsDistanceFromNextSymbol(&symbol_infos);
+
+  ModuleSymbols module_symbols;
+  for (const SymbolInfo& symbol_info : symbol_infos) {
+    *(module_symbols.add_symbol_infos()) = symbol_info;
+  }
```

Comment:
You should be able to move the `SymbolInfo`s:
```suggestion
  for (SymbolInfo& symbol_info : symbol_infos) {
    *(module_symbols.add_symbol_infos()) = std::move(symbol_info);
  }
```